plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+            self.logger.info(f"Total alerts returned: {len(response)}")
+
+            # Local Debugging
+            self.logger.info(f"{type(response) = }")
+            self.logger.info(f"{len(response) = }")
+            self.logger.info(f"{state}, {has_more_pages = }")
+
+            # TODO - Change first None to response
+            # It's set to none so we can see all the logging without the results getting in the way
+
+            # Note - when running locally, we should only get the first 100
+            # When has more pages is True, the task will be rerun automatically again (on staging / the cloud exec)
+
+            # If has more pages == True, then we need to change the search index from 0-100 to 101-200 (Done below now I think)
+            return response, state, has_more_pages, 200, None
+
+        except PluginException as error:
+            self.logger.error(
+                f"A PluginException has occurred. Status code 500 returned. Error: {error.cause}. "
+                f"Existing state: {existing_state}"
+            )
+            return [], existing_state, False, 500, PluginException(cause=error.cause, data=error)
+
+        except Exception as error:
+            print(f"{error = }")
+            return [], existing_state, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
+
+    ###########################
+    # Make request
+    ###########################
+    def get_alerts_palo_alto(self, state, custom_config):
+        endpoint = "/public_api/v1/alerts/get_alerts"
+        response_alerts_field = "alerts"
+        time_sort_field = "creation_time"
+
+        self.logger.info(f"{ALERT_LIMIT = }")
+
+        # IT GOES 0 -> 100 | 101 -> 200 | 201 -> 300
+        # IF NULL, SET FROM = 0
+
+        if not state.get(LAST_SEARCH_FROM):
+            search_from = 0
+        else:
+            # STARTS 1 RECORD AFTER THE LAST ONE PULLED
+            search_from = state.get(LAST_SEARCH_TO) + 1
+            self.logger.info(f"{search_from = }")
+
+        # IF NULL SET TO = 100
+        if not state.get(LAST_SEARCH_TO):
+            search_to = 100
+        else:
+            # SEARCH_TO GETS THE NEWEST SEARCH FROM VALUE AND WILL STOP AFTER PULLING 99 ADDITIONAL RECORDS (100 TOTAL)
+            search_to = search_from + 99
+
+        state[LAST_SEARCH_FROM] = search_from
+        state[LAST_SEARCH_TO] = search_to
+
+        # search_from = state.get(LAST_SEARCH_FROM, 0)
+        # search_to = state.get(LAST_SEARCH_TO, 100) + 100
+
+        # search_to = search_from + ALERT_LIMIT
+        headers = self.connection.xdr_api.get_headers()
+
+        post_body = {
+            "request_data": {
+                "search_from": search_from,
+                "search_to": search_to,
+                "sort": {"field": time_sort_field, "keyword": "asc"},
+            }
+        }
+
+        url = urllib.parse.urljoin("https://api-rapid7.xdr.us.paloaltonetworks.com", endpoint)
```

Comment:
I don't think this url should be hardcoded here...should be coming from the connection or api client instead?